### PR TITLE
Outbound NAT and multiple IPSEC IPs for mobile warriors

### PR DESCRIPTION
**Pull request by Hobby-Student**
- [x] Redmine Issue: https://redmine.pfsense.org/issues/9320
- [x] Ready for review

I think this adds back the default behaviour to all IPSEC clients. Changes were made in #3965 but I haven't looked at the automatic outbound NAT...

### CODE REVIEW

**Comment by Hobby-Student** [comment]
done.

sry for that typo... copy&paste is not always you friend ;)

File: src/etc/inc/filter.inc [hunk 1]
```diff
@@ -1702,6 +1702,19 @@ function filter_nat_rules_automatic_tonathosts($with_descr = false) {
 		$tonathosts[] = "{$config['ipsec']['client']['pool_address']}/{$config['ipsec']['client']['pool_netbits']}";
 		$descriptions[] = gettext("IPsec client");
 	}
+	if ((isset($config['ipsec']['client']['enable'])) &&
```
Comment (rbgarga) [inline src/etc/inc/filter.inc]: Please remove extra parenthesis

File: src/etc/inc/filter.inc [hunk 2]
```diff
@@ -1702,6 +1702,19 @@ function filter_nat_rules_automatic_tonathosts($with_descr = false) {
...
+		(isset($config['ipsec']['mobilekey']))) {
```
Comment (rbgarga) [inline src/etc/inc/filter.inc]: Please remove extra parenthesis

File: src/etc/inc/filter.inc [hunk 3]
```diff
@@ -1702,6 +1702,19 @@ function filter_nat_rules_automatic_tonathosts($with_descr = false) {
...
+		foreach ($config['ipsec']['mobilekey'] as $key) {
+			if (in_array("{$key['pool_address']}/{$key['pool_netbits']}", $tonathosts)) {
+				continue;
+			}
+			if ((!empty($key['pool_address'])) &&
```
Comment (rbgarga) [inline src/etc/inc/filter.inc]: Please remove extra parenthesis

File: src/etc/inc/filter.inc [hunk 4]
```diff
@@ -1702,6 +1702,19 @@ function filter_nat_rules_automatic_tonathosts($with_descr = false) {
...
+				(!empty($key['pool_netbits']))) {
```
Comment (rbgarga) [inline src/etc/inc/filter.inc]: Second level indent should be 4 spaces instead of a tab. While here, please remove extra parenthesis

**Comment by Hobby-Student** [comment]
requested changes were made. hope I got the second level indent right.

**Comment by Hobby-Student** [comment]
just a little update from my side:
have it running on several pfSense boxes. It's working as expected

**Pull request merged**